### ISSUE TITLE
input shape in DDPG

### ISSUE THREAD

**Issue by ffrige** [keras-rl/keras-rl#160]
Hello!
Why is (1,) added to the observation space input both in the actor's and critic's networks for the DDPG agent?
 
actor.add(Flatten(input_shape=(1,) + env.observation_space.shape))

observation_input = Input(shape=(1,) + env.observation_space.shape, name='observation_input')

Can't I just use the env.observation_space.shape by itself?

Thanks!

**Comment by lemonteaa** [comment]
Hi, I think this should be same as #93. The code in DDPG assumes this particular format internally (even if you don't have a window), so you would still need the `(1,)`.

**Comment by ffrige** [comment]
Ah, thank you for pointing me there! I understand the reason for a window size, but we should still be able to remove it if not needed. My actor is a pre-trained policy and does not accept a window size input dimension.

I added a processor and modified the batch in the process_state_batch function to remove the window size. It seems to work fine now.

**Comment by dizcza** [comment]
The issue is in [`select_action()`](https://github.com/keras-rl/keras-rl/blob/master/rl/agents/ddpg.py#L197), where the input dimension is explicitly expanded. I'd also like to have an option not to use window length.
Because of this feature, my model structure becomes clumsy.
Here is an intuitive model I have:

```python
# actor
actor_x_input = Input(shape=observation_space_input_shape, name="observation_input")
actor_x = Dense(8, activation='relu')(actor_x_input)
actor_x = Dense(nb_actions, activation='tanh')(actor_x)
actor = Model(inputs=actor_x_input, outputs=actor_x)

# critic
action_input = Input(shape=(nb_actions,), name='action_input')
observation_input = Input(shape=observation_space_input_shape, name='observation_input')
x = concatenate([action_input, observation_input])
x = Dense(16, activation='relu')(x)
x = Dense(1, activation='linear')(x)
critic = Model(inputs=[action_input, observation_input], outputs=x)
```
and it works well if `state` input is not wrapped in a list (see link above)`. And here is the hack I introduce to expand input and then flatten it:

```python
observation_space_input_shape = (1,) + env.observation_space.shape

# actor
actor_x_input = Input(shape=observation_space_input_shape, name="observation_input")
actor_x = Flatten()(actor_x_input)
actor_x = Dense(8, activation='relu')(actor_x)
actor_x = Dense(nb_actions, activation='tanh')(actor_x)
actor = Model(inputs=actor_x_input, outputs=actor_x)

# critic
action_input = Input(shape=(nb_actions,), name='action_input')
observation_input = Input(shape=observation_space_input_shape, name='observation_input')
flattened_observation = Flatten()(observation_input)
x = concatenate([action_input, flattened_observation])
x = Dense(16, activation='relu')(x)
x = Dense(1, activation='linear')(x)
critic = Model(inputs=[action_input, observation_input], outputs=x)
``` 

**Moreover**, an alternative way of building an actor model produces unexpected error:

```python
actor = Sequential(name="actor")
actor.add(InputLayer(input_shape=observation_space_input_shape, name="observation_input"))
actor.add(Dense(8, activation='relu'))
actor.add(Dense(nb_actions, activation='tanh'))
...
Traceback (most recent call last):
  File "/home/dizcza/PycharmProjects/other/robotics/robotsim/ddpg.py", line 63, in <module>
    agent.compile(Adam(lr=.001), metrics=['mse'])
  File "/home/dizcza/anaconda3/envs/robotsim/lib/python3.7/site-packages/rl/agents/ddpg.py", line 143, in compile
    updates += get_soft_target_model_updates(self.target_actor, self.actor, self.target_model_update)
  File "/home/dizcza/anaconda3/envs/robotsim/lib/python3.7/site-packages/rl/util.py", line 39, in get_soft_target_model_updates
    assert len(target_weights) == len(source_weights)
AssertionError
```
In this case, `target_weights` is an empty list. I spent an hour looking why it happens. The error is so confusing, I'd never understood if I didn't rewrite this example as in the beginning.

P.S. I was thinking to open a new issue, but decided to post it here since it relates much.

Long story short,
the issue of explicit increasing dimension not only requires the user to know this limitation but produces an unexpected error if someone tries to run it differently than constrained but developers.
